base/tps/src/org/dogtagpki/server/tps/TPSTokendb.java

```diff
@@ -783,11 +783,12 @@ private void checkShouldRevoke(TokenRecord tokenRecord, TPSCertRecord cert, Stri
         boolean revokeCerts = configStore.getBoolean(config, true);
 
         if (!revokeCerts) {
-            throw new TPSException(
+            CMS.debug( method +
                     "certificate revocation (serial " + cert.getSerialNumber() +
                     ") not enabled for tokenType: " + tokenType +
                     ", keyType: " + keyType +
                     ", state: " + tokenReason);
+            return;
```

Comment:
Orignally (before commit 33d02d644a1f61bce25698ec28a8ce9890ee3067 was applied) the code would call `tdbActivity()`, then the `shouldRevoke()` and `revokeCert()` would terminate.

After commit 33d02d644a1f61bce25698ec28a8ce9890ee3067 was applied, the `checkShouldRevoke()` would throw an exception (i.e. terminating the execution), then exception handler would swallow the exception and call `tdbActivity()`, then `revokeCert()` would terminate as well, so the behavior was essentially unchanged.

In commit 2f7ed836ab20988386e651c1000f4e12eff6c0af the exception handler was changed to rethrow the exception, which changed the behavior of `revokeCert()`.

In this PR the code is being modified to terminate `checkShouldRevoke()`, but it will not call `tdbActivity()`, and it will continue `revokeCert()`. Is that the expected behavior?

Reply:
Hi @edewata . You are right. Looks like I misread the git show for 33d02d6 .  I think we need to figure out how to differentiate "not wanting to revoke" from "failed to revoke" and treat them differently.
I suppose if we want to keep the isRevoke and revokeExpiredCerts check inside checkShoudRevoke() (to minimize change of code) then we could throw a different kind of Exception from checkShouldRevoke() for if !revokeCert or !revokeExpiredCerts, and then properly swallow it.  Is this okay?  I don't know of any "graceful exit" exception (could create one if we must) or whether this is a good programming idea, but this would introduce less code changes.  I'll go ahead and try to create a new patch in the meantime. Thanks.